chore: bump gradle plugin version

Avishagp opened the pull request:
- [X] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules

#### What does this PR do?
Bumps version of `snyk-gradle-plugin` to 3.16.2 to include this fix (presented through `java-call-graph-builder`): https://github.com/snyk/java-call-graph-builder/pull/57

Use temporary file copied to the local filesystem to ensure the separate process that reads init.gradle can find it in binary releases.

anthogez reviewed (commented):
That’s a “fix: gradle java-call-graph binary” not a chore as it’s introducing changes in the binary releases. Which means that binary users needs to be aware of those changes.

anthogez reviewed (commented):
Also could you include the gh message into the commit’s message body? Thanks!

Commit:
fix: java-call-graph binary

Bumps version of snyk-gradle-plugin to 3.16.2 to include this fix (presented through java-call-graph-builder): snyk/java-call-graph-builder#57

Use temporary file copied to the local filesystem to ensure the separate process that reads init.gradle can find it in binary releases.
